test/ensureArray.js

```diff
@@ -0,0 +1,31 @@
+import { assert } from 'chai';
+import * as RA from '../src/index';
+import eq from './shared/eq';
+
+describe('ensureArray', function() {
+  const emptyArray = [];
+  const filledArray = [42];
+  const arrayInstance = new Array();
+  const arrayPrototype = Array.prototype;
+
+  it('tests `Array` values', function() {
```

Comment:
Can we break  this down into separate tests ?
```js
context('when argument is empty array', function() {
    let emptyArray;

    beforeEach(function() {
      emptyArray = [];
    });

    specify('should return empty array', function() {
      assert.lengthOf(RA.ensureArray(emptyArray, 0));
    });

    specify('should be the same array instance', function() {
      assert.strictEqual(RA.ensureArray(emptyArray, emptyArray));
    });
});
```

And doing this for every case so that we know what we are really testing for.